hermes/src/network/pythnet.rs

```diff
@@ -317,6 +328,115 @@ pub async fn spawn(opts: RunOptions, state: Arc<State>) -> Result<()> {
+                }
+                // This loop with a sleep interval of 1 second allows the task to check for an exit signal at a
+                // fine-grained interval. Instead of sleeping directly for the entire `price_feeds_update_interval`,
+                // which could delay the response to an exit signal, this approach ensures the task can exit promptly
+                // if `crate::SHOULD_EXIT` is set, enhancing the responsiveness of the service to shutdown requests.
+                for _ in 0..price_feeds_update_interval {
+                    if crate::SHOULD_EXIT.load(Ordering::Acquire) {
+                        break;
+                    }
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                }
+            }
+        })
+    };
+
+    let _ = tokio::join!(
+        task_listener,
+        task_guardian_watcher,
+        task_price_feeds_metadata_updater
+    );
     Ok(())
 }
+
+
+pub async fn fetch_and_store_price_feeds_metadata(
+    state: &State,
+    rpc_client: &RpcClient,
+) -> Result<Vec<PriceFeedMetadata>> {
+    let price_feeds_metadata = fetch_price_feeds_metadata(&state, &rpc_client).await?;
+    store_price_feeds_metadata(&state, &price_feeds_metadata).await?;
+    Ok(price_feeds_metadata)
+}
+
+async fn fetch_price_feeds_metadata(
+    state: &State,
+    rpc_client: &RpcClient,
+) -> Result<Vec<PriceFeedMetadata>> {
+    let mut price_feeds_metadata = Vec::<PriceFeedMetadata>::new();
+    let mapping_address = &state.mapping_address;
+    let mapping_data = rpc_client.get_account_data(mapping_address).await?;
+    let mapping_acct = load_mapping_account(&mapping_data)?;
+
+    // Split product keys into chunks of 150 to avoid too many open files error (error trying to connect: tcp open error: Too many open files (os error 24))
+    for product_keys_chunk in mapping_acct
+        .products
+        .iter()
+        .filter(|&prod_pkey| *prod_pkey != Pubkey::default())
+        .collect::<Vec<_>>()
+        .chunks(150)
+    {
+        // Prepare a list of futures for fetching product account data for each chunk
+        let fetch_product_data_futures = product_keys_chunk
+            .iter()
+            .map(|prod_pkey| rpc_client.get_account_data(prod_pkey))
+            .collect::<Vec<_>>();
+
+        // Await all futures concurrently within the chunk
+        let products_data_results = futures::future::join_all(fetch_product_data_futures).await;
+
+        for prod_data_result in products_data_results {
+            match prod_data_result {
+                Ok(prod_data) => {
+                    let prod_acct = match load_product_account(&prod_data) {
+                        Ok(prod_acct) => prod_acct,
+                        Err(e) => {
+                            println!("Error loading product account: {}", e);
+                            continue;
+                        }
+                    };
+
+                    // Ref A
```

Comment:
I think I'm missing the context, what do you mean by this comment?

Reply:
Let's add a ToDo for a stricter typing here.

Reply:
this was added by mistake, removed